api/node/package.json

```diff
@@ -36,11 +36,10 @@
   },
   "scripts": {
     "artifacts": "napi artifacts",
-    "compile": "tsc",
+    "compile": "tsc --build",
     "build": "napi build --platform --release --js rust-module.cjs --dts rust-module.d.ts -c binaries.json",
     "build:debug": "napi build --platform --js rust-module.cjs --dts rust-module.d.ts -c binaries.json && pnpm compile",
     "build:testing": "napi build --platform --js rust-module.cjs --dts rust-module.d.ts -c binaries.json --features testing && pnpm compile",
-    "install": "node build-on-demand.mjs",
```

Comment:
The reason for running this on install is to provide a fallback to build from source when a user installs the package on a machine we don't provide binaries for. This is only needed for packages though, so I thought of keeping it and making it a no-op if .npmignore is present.